Dockerfile

```diff
@@ -1,91 +1,135 @@
+    mix local.rebar --force
 
-COPY assets/package.json assets/package-lock.json $APP_PATH/assets/
-RUN npm install --prefix assets
+# set build ENV
+ENV MIX_ENV="prod"
 
-# Building a release version
-# https://hexdocs.pm/phoenix/releases.html
-FROM elixir_alpine AS production_build
+# install mix dependencies
+COPY mix.exs mix.lock ./
+RUN mix deps.get --only $MIX_ENV
+RUN mkdir config
 
-# Set build ENV
-ENV MIX_ENV=prod
+# copy compile-time config files before we compile dependencies
+# to ensure any relevant config change will trigger the dependencies
+# to be re-compiled.
+COPY config/config.exs config/${MIX_ENV}.exs config/
+RUN mix deps.compile
 
-# Install for dart-sass https://github.com/CargoSense/dart_sass/issues/13
-ARG GLIBC_VERSION=2.33-r0
-RUN wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub
-RUN wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/${GLIBC_VERSION}/glibc-${GLIBC_VERSION}.apk
-RUN apk add glibc-${GLIBC_VERSION}.apk
+COPY priv priv
 
-# Install mix dependencies
-COPY mix.exs mix.lock $APP_PATH/
-RUN mix do deps.get, deps.compile
+COPY lib lib
 
-# Compile and build release
-COPY . .
+COPY assets assets
 
-# Build assets
-# COPY assets/package.json assets/package-lock.json $APP_PATH/assets/
+# Install npm packages:
+COPY assets/package.json assets/package-lock.json ./assets/
 RUN npm install --prefix assets
+
+# compile assets
 RUN mix assets.deploy
-RUN mix do phx.digest, compile, release
 
-# Prepare release image
-FROM alpine:${ALPINE_VERSION} AS production
+# Compile the release
+RUN mix compile
+
+# Changes to config/runtime.exs don't require recompiling the code
+COPY config/runtime.exs config/
+
+COPY rel rel
+RUN mix release
 
-ENV APP_PATH=/app
+# start a new build stage so that the final image will only contain
+# the compiled release and other runtime necessities
+FROM ${RUNNER_IMAGE} as production
 
-RUN apk add --no-cache \
-      libgcc \
-      libstdc++ \
-      ncurses-libs \
-      openssl \
-      postgresql-client \
-    ;
+RUN apt-get update -y && apt-get install -y libstdc++6 openssl libncurses5 locales \
+  && apt-get clean && rm -f /var/lib/apt/lists/*_*
 
-WORKDIR $APP_PATH
+# Set the locale
+RUN sed -i '/en_US.UTF-8/s/^# //g' /etc/locale.gen && locale-gen
 
-RUN chown nobody:nobody $APP_PATH/
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
 
-USER nobody:nobody
+WORKDIR "/app"
+RUN chown nobody /app
 
-COPY .docker/entrypoint.release.sh $APP_PATH/.docker/entrypoint.release.sh
-COPY --from=production_build --chown=nobody:nobody $APP_PATH/_build/prod/rel/qrstorage ./
+# set runner ENV
+ENV MIX_ENV="prod"
 
-ENV HOME=$APP_PATH
+# Only copy the final release from the build stage
+COPY --from=builder --chown=nobody:root /app/_build/${MIX_ENV}/rel/qrstorage ./
 
-ENTRYPOINT ["sh", ".docker/entrypoint.release.sh"]
+USER nobody
 
-EXPOSE 8000
\ No newline at end of file
+CMD ["/app/bin/server"]
```

Comment:
Attention: This startup script does not execute migrations as it was done before. The previous start up script `.docker/entrypoint.release.sh` executed also the migrations.